### PR TITLE
Add Resources section to README

### PR DESCRIPTION
**Pull request by ejmastnak**
Two changes:
- Added a section of third-party resources to the README
- Rearranged the official documentation options into a (hopefully) more readable bulleted list

refer: discussion #545

### CODE REVIEW

**Comment by ejmastnak** [comment]
Good idea, agreed! The resources are probably a better place to get started as a beginner than DOC.md. I’ll tweak the README when I get home.

**Comment by ejmastnak** [comment]
Just added a first attempt at a rewording, let me know what you think and feel free to change/rearrange!

**Comment by L3MON4D3** [comment]
Riiight, I can also push to your PR-branch.
I'll make some changes and summarize them here, easier to discuss I think:
* split new users-section into lua and vscode/snipmate: If one only needs very simple snippets, it makes more sense to head to the snipmate/vscode-loaders
* Replace linear approach (learn this -> this -> this) with basics -> several advanced/need to know topics: I think this makes more sense, after grokking the basics, there's no real order to the advanced stuff
* Add note about `ls.snippets = ` to `add_snippets`-transition (tjs videos were recorded before we made that significant change)

**Comment by ejmastnak** [comment]
Nice! I've added a few capitalization tweaks and upgraded the Getting Started section to its own heading but otherwise no major changes from your last commit and totally agree with the non-linear learning structure and Snipmate/Lua division.

I agree learning the advanced features in parallel makes more sense (instead of this -> this -> this), come to think of it this is how I learned myself haha. And good catch about the `ls.snippets = ` to `add_snippets`-transition in TJs video!

File: README.md [hunk 1]
```diff
@@ -99,32 +99,40 @@ This can also be done much cleaner, with all the benefits that come with using a
...
+  Of those two, SnipMate is definitely the more comfortable way of writing snippets.
+* **Lua snippets**: we suggest first watching or reading one of the introductory guides in the [Resources for new users](#resources-for-new-users) section below.
+  After getting familiar with the basics, you should check out the important LuaSnip features in the following list:
+  * [`config`](https://github.com/L3MON4D3/LuaSnip#config): Notable: `region_check_events` for jumping to the end of snippets the cursor is no longer inside of,
+    `delete_check_events` for cleaning up snippets whose text was deleted,
+    and `enable_autosnippets` to enable automatic snippet expansion.
   * [`extras`](https://github.com/L3MON4D3/LuaSnip/blob/master/DOC.md#extras): This module contains many functions that make writing snippets
-    significantly easier.  
+    significantly easier;
```
Comment (L3MON4D3) [inline README.md:116]: fancy :ok_hand: :D

**Comment by L3MON4D3** [comment]
This looks perfect :+1:
Squash -> Merge?

**Comment by ejmastnak** [comment]
> This looks perfect +1 Squash -> Merge?

Great! Shall I squash locally and then force push upstream or do you want to take over from here and squash/merge this PR directly from GitHub?

**Comment by L3MON4D3** [comment]
Mhm I haven't tried github squash+merge yet, lemme see

**Comment by L3MON4D3** [comment]
Looks good I think.
Thank you very much👍👍

**Pull request merged**